.github/workflows/test.yml

```diff
@@ -11,23 +11,28 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        go-version: [1.17.x, 1.18.x]
-        os: [ubuntu-latest, macos-latest, windows-latest]
+        go-version:
+        - stable
+        - oldstable
```

Comment:
I'm personally not a fan of this, because in my opinion we want to be conscious about what major versions of Go we support.

We do tend to do the last two versions, but when a new version comes out, it's a human process to update the versions in multiple places (CI, go.mod, etc) and test that everything still works. Sometimes a few tests break, especially since we have tight coupling with some packages like testing - and we also don't want CI to start failing when a new version of Go is out :)

but perhaps @rogpeppe disagrees. Some people prefer their CI to track "latest stable" with the Go version, and then they can react when CI starts failing on its own. I personally prefer to keep CI closer to determinism, so that it's not likely to start failing on its own.

Reply:
Worth noting that depending on the project I also track versions of the operating systems manually. For example, on a project of mine, `macos-latest` broke a test when that jumped from `macos-11` to `macos-12`. But, for most Go projects, OS upgrades don't change behavior.

Reply:
Ack, I've changed the PR to use Go versions `1.18.x` and `1.19.x` explicitly.